src/coreclr/jit/fgopt.cpp

```diff
@@ -145,16 +145,16 @@ bool Compiler::fgReachable(BasicBlock* b1, BasicBlock* b2)
 
     if (b1->bbNum > fgDomBBcount)
     {
-        noway_assert(b1->bbJumpKind == BBJ_NONE || b1->bbJumpKind == BBJ_ALWAYS || b1->bbJumpKind == BBJ_COND);
+        noway_assert(b1->KindIs(BBJ_NONE, BBJ_ALWAYS, BBJ_COND));
 
-        if (b1->bbFallsThrough() && fgReachable(b1->bbNext, b2))
+        if (b1->KindIs(BBJ_NONE, BBJ_COND) && fgReachable(b1->bbNext, b2))
```

Comment:
Doesn't this change behavior? `bbFallsThrough()` checks a flag for `BBJ_CALLFINALLY`.

Reply:
We already `noway_assert` for it being NONE/ALWAYS/COND, so we won't see a BBJ_CALLFINALLY here.